src/v/security/jwt.h

```diff
@@ -0,0 +1,623 @@
+concept string_viewable = detail::is_string_viewable<T>::value;
+
+template<typename ToCharT = char>
+std::basic_string_view<ToCharT> char_view_cast(string_viewable auto const& sv) {
+    return {reinterpret_cast<ToCharT const*>(sv.data()), sv.length()};
+}
+
+struct string_viewable_compare {
+    using is_transparent = void;
+    bool operator()(
+      string_viewable auto const& lhs, string_viewable auto const& rhs) const {
+        return char_view_cast(lhs) == char_view_cast(rhs);
+    }
+};
+
+struct string_viewable_hasher {
+    using is_transparent = void;
+    size_t operator()(string_viewable auto const& sv) const {
+        return std::hash<std::string_view>{}(char_view_cast(sv));
+    }
+};
+
+using cryptopp_bytes = ss::basic_sstring<CryptoPP::byte, uint32_t, 31, false>;
+using cryptopp_byte_view = std::basic_string_view<cryptopp_bytes::value_type>;
+
+template<typename CharT = std::string_view::value_type>
+std::basic_string_view<CharT> as_string_view(json::Value const& v) {
+    expression_in_debug_mode(
+      vassert(
+        v.IsString(), "only Strings can be converted to std::string_view"););
+    return {reinterpret_cast<CharT const*>(v.GetString()), v.GetStringLength()};
+}
+
+template<typename CharT = std::string_view::value_type>
+std::optional<std::basic_string_view<CharT>>
+string_view(json::Value const& doc, std::string_view field) {
+    auto it = doc.FindMember(field.data());
+    if (it == doc.MemberEnd() || !it->value.IsString()) {
+        return std::nullopt;
+    }
+    return as_string_view<CharT>(it->value);
+}
+
+template<string_viewable StringT = cryptopp_bytes>
+auto base64_url_decode(cryptopp_byte_view sv) {
+    CryptoPP::Base64URLDecoder decoder;
+
+    decoder.Put(sv.data(), sv.size());
+    decoder.MessageEnd();
+
+    StringT decoded;
+    CryptoPP::word64 size = decoder.MaxRetrievable();
+    if (size && size <= SIZE_MAX) {
+        decoded.resize(size);
+        decoder.Get(
+          reinterpret_cast<CryptoPP::byte*>(decoded.data()), decoded.size());
+    }
+    return decoded;
+};
+
+inline std::optional<CryptoPP::Integer>
+as_cryptopp_integer(json::Value const& v, std::string_view field) {
+    CryptoPP::Integer integer;
+    auto b64 = string_view<CryptoPP::byte>(v, field);
+    if (!b64.has_value()) {
+        return std::nullopt;
+    }
+    auto chars = base64_url_decode(b64.value());
+    return CryptoPP::Integer{chars.data(), chars.size()};
+}
+
+auto check_bytes(detail::string_viewable auto const& s) {
+    auto sv = detail::char_view_cast<char>(s);
+    return !sv.empty() && is_valid_utf8(sv) && !contains_control_character(sv);
+};
+
+template<typename Clock>
+std::optional<typename Clock::time_point>
+time_point(json::Value const& doc, std::string_view field) {
+    auto it = doc.FindMember(field.data());
+    if (it == doc.MemberEnd() || !it->value.IsInt64()) {
+        return std::nullopt;
+    }
+    return
+      typename Clock::time_point(std::chrono::seconds(it->value.GetInt64()));
```

Comment:
For `exp`, `iat` and `nbf`, the value is defined as a [NumericDate](https://www.rfc-editor.org/rfc/rfc7519#section-2):
```
A JSON numeric value representing the number of seconds from
      1970-01-01T00:00:00Z UTC until the specified UTC date/time,
      ignoring leap seconds.  This is equivalent to the IEEE Std 1003.1,
      2013 Edition [[POSIX.1](https://www.rfc-editor.org/rfc/rfc7519#ref-POSIX.1)] definition "Seconds Since the Epoch", in
      which each day is accounted for by exactly 86400 seconds, other
      than that non-integer values can be represented.  See [RFC 3339](https://www.rfc-editor.org/rfc/rfc3339)
      [[RFC3339](https://www.rfc-editor.org/rfc/rfc3339)] for details regarding date/times in general and UTC in
      particular.